src/components/progress-tracker/progress-tracker.pw.tsx

```diff
@@ -0,0 +1,450 @@
+import { expect, test } from "@playwright/experimental-ct-react17";
+import React from "react";
+
+import {
+  progressTrackerComponent,
+  progressTrackerCustomValuePreposition,
+  progressTrackerDescription,
+  progressTrackerLine,
+  progressTrackerMaxVal,
+  progressTrackerMinVal,
+} from "../../../playwright/components/progress-tracker";
+import { CHARACTERS } from "../../../playwright/support/constants";
+import {
+  assertCssValueIsApproximately,
+  checkAccessibility,
+  checkCSSOutline,
+} from "../../../playwright/support/helper";
+import { ProgressTrackerProps } from "../progress-tracker";
+import { ProgressTrackerComponent } from "./components.test-pw";
+import { PROGRESS_TRACKER_SIZES } from "./progress-tracker.config";
+
+const DEFAULT_PROP_VALUE = 50;
+
+const checkPropName = (propName: string) =>
+  propName.endsWith("now") &&
+  propName.endsWith("min") &&
+  propName.endsWith("max");
+
+const getProps = (propName: string, shouldBeDefault: boolean) => {
+  if (!shouldBeDefault) {
+    return {
+      [propName]: DEFAULT_PROP_VALUE,
+    };
+  }
+  return {
+    [propName]: CHARACTERS.STANDARD,
+  };
+};
+
+test.describe("Check props for ProgressTracker component", () => {
+  ([
+    "aria-label",
+    "aria-describedby",
+    "aria-valuenow",
+    "aria-valuemin",
+    "aria-valuemax",
+    "aria-valuetext",
+  ] as const).forEach((propName) => {
+    test.describe(`When ${propName} prop is passed`, () => {
+      test(`verify that the ${propName} is set to cypress-standard`, async ({
+        mount,
+        page,
+      }) => {
+        const isNowMinOrMax = checkPropName(propName);
+        const props = getProps(propName, isNowMinOrMax);
+        const assertion = isNowMinOrMax
+          ? CHARACTERS.STANDARD
+          : DEFAULT_PROP_VALUE;
+
+        await mount(<ProgressTrackerComponent {...props} />);
+
+        const progressTrackerComponentElement = progressTrackerComponent(page);
+        await expect(progressTrackerComponentElement).toHaveAttribute(
+          propName,
+          String(assertion)
+        );
+      });
+    });
+  });
```

Comment:
**suggestion**: what do you think about passing the expected value for each ARIA prop as a test parameter?

```tsx
  ([
    ["aria-label", CHARACTERS.STANDARD],
    ["aria-describedby", CHARACTERS.STANDARD],
    ["aria-valuenow", 50],
    ["aria-valuemin", 50],
    ["aria-valuemax", 50],
    ["aria-valuetext", CHARACTERS.STANDARD],
  ] as const).forEach(([propName, value]) => {
    test(`when ${propName} prop is passed, sets the correct ARIA attribute on the ProgressTracker`, async ({
      mount,
      page,
    }) => {
      await mount(<ProgressTrackerComponent {...{ [propName]: value }} />);

      const progressTrackerComponentElement = progressTrackerComponent(page);
      await expect(progressTrackerComponentElement).toHaveAttribute(
        propName,
        String(value)
      );
    });
```

That way we could eliminate the `checkPropName` and `getProps` functions to improve readability.